posthog/hogql/printer.py

```diff
@@ -199,95 +301,208 @@ def print_ast(
                 )
 
             # check that we're not running inside another aggregate
-            for stack_node in stack:
+            for stack_node in self.stack:
                 if stack_node != node and isinstance(stack_node, ast.Call) and stack_node.name in HOGQL_AGGREGATIONS:
                     raise ValueError(
                         f"Aggregation '{node.name}' cannot be nested inside another aggregation '{stack_node.name}'."
                     )
 
-            translated_args = ", ".join([print_ast(arg, stack, context, dialect) for arg in node.args])
-            if dialect == "hogql":
-                response = f"{node.name}({translated_args})"
+            translated_args = ", ".join([self.visit(arg) for arg in node.args])
+            if self.dialect == "hogql":
+                return f"{node.name}({translated_args})"
             elif node.name == "count":
-                response = "count(*)"
+                return "count(*)"
+            # TODO: rework these
             elif node.name == "countDistinct":
-                response = f"count(distinct {translated_args})"
+                return f"count(distinct {translated_args})"
```

Comment:
not a big fan of these translations, why do we have them?

Reply:
We're going to get rid of them, but outside of this PR 😅 

https://github.com/PostHog/meta/issues/81

> [ ] Clean up the confusion with our bespoke countDistinct() function

They're leftovers from the Python AST days. I want to support native ANSI-compliant SQL here.